how-to/software/snapshot-service.md

```diff
@@ -0,0 +1,284 @@
+  environments or by rolling out package upgrades in stages for a large set of
+  machines.
+
+This how-to guide shows how to set-up and use the snapshot service.
+
+## Prerequisites
+
+Snapshots are supported in Ubuntu 23.10 onwards and on updated installations of
+Ubuntu 20.04 LTS (starting from `apt` 2.0.10) and Ubuntu 22.04 LTS (starting
+from `apt` 2.4.11).
+
+In Ubuntu 24.04 LTS and later, `apt` will automatically detect whether a
+repository supports snapshots. Therefore, there is no need for additional
+configuration before you can start using the Ubuntu snapshot service. For
+previous versions, you need to set a `snapshot` option in the relevant entries
+of the `/etc/apt/sources.list` file. For instance:
+
+If you have the following contents in the `/etc/apt/sources.list` file
+
+```
+deb http://archive.ubuntu.com/ubuntu jammy main universe restricted multiverse
+deb http://archive.ubuntu.com/ubuntu jammy-updates main universe restricted multiverse
+deb http://archive.ubuntu.com/ubuntu jammy-backports main restricted universe multiverse
+deb http://security.ubuntu.com/ubuntu jammy-security main restricted universe multiverse
+```
+
+you can enable the snapshot service for all components in all pockets by changing it to
+
+```
+deb [snapshot=yes] http://archive.ubuntu.com/ubuntu jammy main universe restricted multiverse
+deb [snapshot=yes] http://archive.ubuntu.com/ubuntu jammy-updates main universe restricted multiverse
+deb [snapshot=yes] http://archive.ubuntu.com/ubuntu jammy-backports main restricted universe multiverse
+deb [snapshot=yes] http://security.ubuntu.com/ubuntu jammy-security main restricted universe multiverse
+```
+
+Without the setup shown above, `apt` will simply ignore snapshot related
+command line options in Ubuntu releases lower than 24.04.
+
+## Using the snapshot service
+
+There are several options when it comes to setting up Ubuntu snapshot services.
+Here we present the most relevant, supported ones.
+
+Below, we refer to the Snapshot ID as `$SNAPSHOT_ID`. This is a UTC date and
+time formatted as YYYYMMDDTHHMMSSZ, e.g., 20240431T214500Z, which refers to the
+snapshot that represents the state of the Ubuntu archive on April 31st, 2024,
+at 21:45:00 UTC.
+
+Currently, the Ubuntu snapshot service provides snapshots for any date and time
+**after 1 March 2023**.
+
+### The `--snapshot` CLI option
+
+The snapshot service can be used through the `--snapshot` (or `-S`) CLI option
+for `apt`, which receives a Snapshot ID argument.
+
+```
+apt update --snapshot $SNAPSHOT_ID
+apt install $PACKAGE_NAME --snapshot $SNAPSHOT_ID
+```
+
+you can also use the short, atomic version to merge the two actions above in a
+single command, which will download the package information from the
+repository, download, and install the package:
+
+```
+apt install $PACKAGE_NAME --update --snapshot $SNAPSHOT_ID
+```
+
+or using the short version of the snapshot option:
+
+```
+apt install $PACKAGE_NAME --update --S $SNAPSHOT_ID
```

Comment:
todo: Fix short version with single dash `-S` and consider just showing the short version, to avoid duplication. The `--snapshot` variant is already displayed above.
```suggestion
apt install $PACKAGE_NAME --update -S $SNAPSHOT_ID
```

Reply:
OK then, how about I leave this one as is (to show the usage of the short version of the command) and replace any other `-S` occurrences to `--snapshot`